pkg/adsc/adsc.go

```diff
@@ -934,12 +938,13 @@ func (a *ADSC) handleEDS(eds []*endpoint.ClusterLoadAssignment) {
 		b, _ := json.MarshalIndent(eds, " ", " ")
 		adscLog.Debugf(string(b))
 	}
-	if a.InitialLoad == 0 {
+	if a.initialLoad == 0 && !a.initialLds {
 		// first load - Envoy loads listeners after endpoints
 		_ = a.stream.Send(&discovery.DiscoveryRequest{
 			Node:    a.node(),
 			TypeUrl: v3.ListenerType,
 		})
+		a.initialLds = true
```

Comment:
It looks like the same effect as moving `a.InitialLoad = time.Since(a.watchTime)` here.

In addition, it may not be safe: when sending LDS or RDS encounters an unexpected situation (note that we do not handle sending errors), there may never be a chance to resend LDS?

Reply:
> I am not sure what do you mean.

see L980 https://github.com/istio/istio/pull/49200/files#diff-b5f677f06a05070210449a3f61f748b01824b494d5cf4116f88e50606c621d47R980

Reply:
should i do the error handling in a following work, the framework need some refactor

Reply:
Please wait for the maintainer review and ignore my comment.